driver-core/src/main/com/mongodb/client/model/expressions/Branches.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.model.expressions;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+
+public final class Branches {
+
+    Branches() {
+        // package-private
+    }
+
+    private <T extends Expression, R extends Expression> BranchesIntermediary<T, R> with(final Function<T, SwitchCase<R>> switchCase) {
+        List<Function<T, SwitchCase<R>>> v = new ArrayList<>();
+        v.add(switchCase);
+        return new BranchesIntermediary<>(v);
+    }
+
+    private static <T extends Expression> MqlExpression<?> mqlEx(final T value) {
+        return (MqlExpression<?>) value;
+    }
+
+    // is fn
+
+    public <T extends Expression, R extends Expression> BranchesIntermediary<T, R> is(final Function<T, BooleanExpression> o, final Function<T, R> r) {
```

Comment:
I believe I have made all of these changes in the relevant places (though the params varied).

Reply:
The following places still need a change:

- `Branches`/`BranchesIntermediary.is[Type]` need `Function<? super [Type], ...>` instead of `Function<[Type], ...>`.
- `BranchesIntermediary.is` needs `Function<? super T, BooleanExpression>` instead of `Function<T, BooleanExpression>`.